### PR TITLE
Improve repr

### PR DESCRIPTION
**Pull request by teo-milea**
## Description
Made some changes for different objects' repr:
* added `__attr_custom_repr__` to `SyftObject` to overwrite the attrs we want to print for an object of that type
* limited `Request` repr to:
  * status
  * requesting_user_verify_key
  * approval time
  * request time
  * changes
* limited `Project` repr to:
  *  id
  * name
  * description
  * user_email_address
  * events
  * user_signing_key
* limited `ProjectSubmit` repr to:
  *  id
  * name
  * description
  * user_email_address
* Added a force parameter to accept_by_depositing_result to overwrite an already accepted result
* Shorten type and uid when printing a list of objects in notebooks
* Removed limit when rendering pandas dataframe
* Added itables for pagination for list/dicts

Also merge #7711 and #7712.

* [TODO] Improve the repr for Dataset and Assets with Markdown  

## How has this been tested?
- Notebook testing / updated notebooks
- unit tests

## UX Changes:
* shorten UIDs and types
image
* added pagination with itables (needs styling)
image
* added `__attr_custom_repr__` for custom single object repr
image
image
* added `force` parameter in `.accept_by_depositing_result` for result overloading
image

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: packages/syft/src/syft/service/request/request.py [hunk 1]
```diff
@@ -238,18 +245,33 @@ def accept_by_depositing_result(self, result: Any):
...
+            if not force:
+                return SyftError(
+                    message="Already approved, if you want to force updating the result use force=True"
+                )
+            action_obj_id = state.output_history[0].outputs[0]
+            action_object = ActionObject.from_obj(result, id=action_obj_id)
+            result = api.services.action.save(action_object)
+            if not result:
+                return result
+            return SyftSuccess(message="Request submitted for updating result.")
```
Comment (koenvanderveen) [inline packages/syft/src/syft/service/request/request.py:269]: maybe add an else here for readability?

File: packages/syft/src/syft/service/project/project.py [hunk 2]
```diff
@@ -657,6 +657,14 @@ def add_code_request_to_project(
 class Project(SyftObject):
     __canonical_name__ = "Project"
     __version__ = SYFT_OBJECT_VERSION_1
+    __attr_custom_repr__ = [
+        "id",
+        "name",
+        "description",
+        "user_email_address",
+        "events",
+        "user_signing_key",
```
Comment (rasswanth-s) [inline packages/syft/src/syft/service/project/project.py:666]: Could we remove signing key from repr, as it is a sensitive credential, we could avoid printing it?

Comment (rasswanth-s) [reply packages/syft/src/syft/service/project/project.py:666]: Sorry, it is my mistake, I looked it in ticket, I found that, I incorrectly mentioned signing key.

Comment (rasswanth-s) [reply packages/syft/src/syft/service/project/project.py]: Resolved in [4c7f7db](https://github.com/OpenMined/PySyft/pull/7717/commits/4c7f7dba1511442c166a9796122bdb4ab13c0ff0)

**Review by koenvanderveen — APPROVED** [review]

**Pull request merged**